source/Octopus.Server.Client/Operations/IRegisterWorkerOperation.cs

```diff
@@ -12,6 +10,12 @@ public interface IRegisterWorkerOperation : IRegisterMachineOperationBase
         /// <summary>
         /// Gets or sets the worker pools that this machine should be added to.
         /// </summary>
+        [Obsolete($"Use the {nameof(WorkerPools)} property as it supports worker pool names, slugs and Ids.")]
```

Comment:
The people using this library will get compiler warnings about its deprecation. I think this is enough because the only way this can get used is through code.